### ISSUE TITLE
finetune on KITTI details

### ISSUE THREAD

**Issue by Guptajakala** [Windsrain/Selective-Stereo#5]
Hi, can you share more details on the finetuning on KITTI? E.g. what is the learning rate schedule, max lr, image size, data augmentations?

**Comment by Windsrain** [comment]
You can view Sec. 4.3 in our paper. Except as mentioned directly, you can keep default settings. Besides, I suggest that you save checkpoints per 1k steps or even smaller to choose the best.

**Comment by Guptajakala** [comment]
Thanks, that's very helpful. 

>to choose the best

Do you have some train/val split on KITTI that you defined to choose the best validation result? Could you share it?

**Comment by Windsrain** [comment]
> Thanks, that's very helpful.
> 
> > to choose the best
> 
> Do you have some train/val split on KITTI that you defined to choose the best validation result? Could you share it?

At first, I imitated GwcNet to partition the validation set. Now I just test checkpoints and choose the best on the leaderboard, because there is a certain degree of randomness due to the small size of KITTI. Besides, if you want to get better results, I suggest that you first train the sceneflow model with the VKITTI2 dataset, and then finetune it on KITTI. I think doing so will get better results than our paper.

**Comment by Guptajakala** [comment]
Thanks for your great suggestion!! Do you have a reference for VKITTI2? I searched it but cannot find it.

**Comment by Windsrain** [comment]
> Thanks for your great suggestion!! Do you have a reference for VKITTI2? I searched it but cannot find it.

You can get the link here https://github.com/autonomousvision/unimatch.

View DATASETS.md and view its dataset codes to get disparity maps.

**Comment by Guptajakala** [comment]
thank you so much on this!

**Comment by S1aoXuan** [comment]
> > Thanks, that's very helpful.
> > > to choose the best
> > 
> > 
> > Do you have some train/val split on KITTI that you defined to choose the best validation result? Could you share it?
> 
> At first, I imitated GwcNet to partition the validation set. Now I just test checkpoints and choose the best on the leaderboard, because there is a certain degree of randomness due to the small size of KITTI. Besides, if you want to get better results, I suggest that you first train the sceneflow model with the VKITTI2 dataset, and then finetune it on KITTI. I think doing so will get better results than our paper.

作者你好，我想问问在如何去选择最好的checkpoint呢？KITTI的测试集没有真值，现在KITTI官方一个账号一个月也只能提交3次了。希望您能给一些指导

**Comment by Windsrain** [comment]
1. 减小保存间隔；
2. 因为是在混合KITTI上训练，所以分别针对2012/2015，可以直接用对应的训练集先看看指标，太差的指标没必要提交；

正常来说就两点，因为我写论文有合作者，所以他们也可以帮我测试。
测试其实是在不断完善的。一开始PSM和Gwc是划分验证集来挑，但问题在于KITTI数据本来就少，再划分验证集就更少了，所以从ACVNet开始就用混合KITTI数据集训练。最近一些论文如unimatch也开始使用虚拟数据集如VKITTI2来减少随机性。其实我觉得之后的文章也会慢慢采用unimatch这种方式。

**Comment by S1aoXuan** [comment]
> 1. 减小保存间隔；
> 2. 因为是在混合KITTI上训练，所以分别针对2012/2015，可以直接用对应的训练集先看看指标，太差的指标没必要提交；
> 
> 正常来说就两点，因为我写论文有合作者，所以他们也可以帮我测试。 测试其实是在不断完善的。一开始PSM和Gwc是划分验证集来挑，但问题在于KITTI数据本来就少，再划分验证集就更少了，所以从ACVNet开始就用混合KITTI数据集训练。最近一些论文如unimatch也开始使用虚拟数据集如VKITTI2来减少随机性。其实我觉得之后的文章也会慢慢采用unimatch这种方式。

谢谢！

**Comment by Guptajakala** [comment]
@Windsrain how do you select the ckpt for Middlebury? I have never submitted to it before. Are you able to do the similar things as KITTI?

**Comment by Windsrain** [comment]
> @Windsrain how do you select the ckpt for Middlebury? I have never submitted to it before. Are you able to do the similar things as KITTI?

No, I just select the last ckpt for Middlebury, but you can do the similar things as KITTI. For example, you can select ckpt based on the quota of the training set. You can also select it based on the visualized observations of the test set.
